src/pages/receipts/packagesInTransit.js

```diff
@@ -534,8 +513,42 @@ const addManifestTableRows = (boxNumber, bagIdArr, index, groupSamples, groupSca
 
 const tempProbeFound = (tempProbe) => {
     const options = {
-      '104430631': 'No',
-      '353358909': 'Yes'
+        '104430631': 'No',
+        '353358909': 'Yes'
     };
     return options[tempProbe] || '';
-  };
\ No newline at end of file
+};
+
+/**
+ * Creates the object packagesInTransitModalData and stores it in state
+ * @param {object} packagesInTransitDataObject - stored object in state with array values for each key, ex. { sumSamplesArr: [], bagSamplesArr: [], ... } 
+ * @param {number} index - index of modal button 
+ * @param {array} allBoxesShippedBySiteAndNotReceived - array of objects with all boxes shipped by site and not received sorted by most recent data
+*/
+
+const savePackagesInTransitModalData = (packagesInTransitDataObject, index, allBoxesShippedBySiteAndNotReceived ) => {
+    const { bagSamplesArr, shippedByArr, trackingNumberArr, groupByLoginSiteCidArr } = packagesInTransitDataObject
+    const { shippingShipDate, shippingLocation, shippingBoxId } = fieldToConceptIdMapping;
+    
+    let packagesInTransitModalData = {
+        site: "",
+        date: "",
+        location: "",
+        boxNumber: "",
+        groupSamples: "",
+        groupShippedBy: "",
+        trackingNumber: "",
+        loginSite: "",
+    }
+
+    packagesInTransitModalData.site = allBoxesShippedBySiteAndNotReceived[index].siteAcronym;
+    packagesInTransitModalData.date = allBoxesShippedBySiteAndNotReceived[index][shippingShipDate];
+    packagesInTransitModalData.location = allBoxesShippedBySiteAndNotReceived[index][shippingLocation];
+    packagesInTransitModalData.boxNumber = allBoxesShippedBySiteAndNotReceived[index][shippingBoxId];
+    packagesInTransitModalData.groupSamples = bagSamplesArr[index];
+    packagesInTransitModalData.groupShippedBy = shippedByArr;
+    packagesInTransitModalData.trackingNumber = trackingNumberArr[index];
+    packagesInTransitModalData.loginSite = groupByLoginSiteCidArr[index];
+
+    appState.setState({ packagesInTransitModalData: packagesInTransitModalData });
```

Comment:
`{packagesInTransitModalData}` should do it.

Reply:
I see, I'll modify it to object property shorthand.